modules/form-completion-status/src/main/java/io/uhndata/cards/formcompletionstatus/AnswerCompletionStatusEditor.java

```diff
@@ -107,11 +109,16 @@ public void propertyChanged(final PropertyState before, final PropertyState afte
         if (questionNode != null && PROP_VALUE.equals(after.getName())) {
             final Iterable<String> nodeAnswers = after.getValue(Type.STRINGS);
             final int numAnswers = iterableLength(nodeAnswers);
-            final List<String> statusFlags = new ArrayList<>();
+            final Set<String> statusFlags = new TreeSet<>();
+            if (before != null) {
+                before.getValue(Type.STRINGS).forEach(statusFlags::add);
+            }
```

Comment:
Ah, this has two bugs:

- the `before` state is the old value, not the one you're trying to set, which means that trying to set new flags will always go back to the previous state, plus or minus INVALID/INCOMPLETE, but without any of the new values; `after` should be used instead, but:
- the `before` and `after` are only about the `value` property in here, not the status flags, which are only found in the `currentNodeBuilder`

This should work:

```suggestion
            this.currentNodeBuilder.getProperty(STATUS_FLAGS).getValue(Type.STRINGS).forEach(statusFlags::add);
```